### PR TITLE
Rewrite style props

### PR DESCRIPTION
**Pull request by rmorshea**
## Checklist

Please update this checklist as you complete each item:

- [x] Tests have been included for all bug fixes or added functionality.
- [x] The `changelog.rst` has been updated with any significant changes.
- [ ] GitHub Issues which may be closed by this Pull Request have been linked.

### CODE REVIEW

File: docs/source/reference/_examples/simple_dashboard.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def update_value(value):
         set_value_callback(value)
 
     return idom.html.fieldset(
-        {"class": "number-input-container"},
+        {"class_": "number-input-container"},
```
Comment (Archmonger) [inline docs/source/reference/_examples/simple_dashboard.py]: Is this correct?

Comment (rmorshea) [reply docs/source/reference/_examples/simple_dashboard.py]: Nope, good catch.

**Review by Archmonger — APPROVED** [review]

**Comment by Archmonger** [comment]
Do we want to tackle the [`custom_vdom_constructor` pylint issues](https://github.com/idom-team/idom/pull/919#issuecomment-1441081230)?

**Comment by rmorshea** [comment]
So `custom_vdom_constructor` actually does implement an overload. My suspicion is that `pylint` assumes decorators don't modify function interfaces - hence the error messages. IMO, if you're using MyPy you don't need PyLint. If you want to continue using it, I'd recommend just turning off this check.

**Comment by Archmonger** [comment]
Pylint catches a lot of things that I've seen mypy miss. Typically that's a fairly useful error to retain.

It seems like a bug within Pylint, and the generally accepted solution is to directly call the decorator, rather than using the `@` decorator operation.

https://stackoverflow.com/questions/40135129/pylint-getting-it-to-understand-decorators

**Pull request merged**